glib/src/match_info.rs

```diff
@@ -1,9 +1,380 @@
+impl<'input> FromGlibPtrNone<*mut ffi::GMatchInfo> for MatchInfo<'input> {
+    #[inline]
+    unsafe fn from_glib_none(ptr: *mut ffi::GMatchInfo) -> Self {
+        Self {
+            inner: from_glib_none(ptr),
+            _phantom: std::marker::PhantomData,
+        }
+    }
+}
+#[doc(hidden)]
+impl<'input> FromGlibPtrNone<*const ffi::GMatchInfo> for MatchInfo<'input> {
+    #[inline]
+    unsafe fn from_glib_none(ptr: *const ffi::GMatchInfo) -> Self {
+        Self {
+            inner: from_glib_none(ptr),
+            _phantom: std::marker::PhantomData,
+        }
+    }
+}
+#[doc(hidden)]
+impl<'input> FromGlibPtrFull<*mut ffi::GMatchInfo> for MatchInfo<'input> {
+    #[inline]
+    unsafe fn from_glib_full(ptr: *mut ffi::GMatchInfo) -> Self {
+        Self {
+            inner: from_glib_full(ptr),
+            _phantom: std::marker::PhantomData,
+        }
+    }
+}
+#[doc(hidden)]
+impl<'input> FromGlibPtrBorrow<*mut ffi::GMatchInfo> for MatchInfo<'input> {
+    #[inline]
+    unsafe fn from_glib_borrow(ptr: *mut ffi::GMatchInfo) -> Borrowed<Self> {
+        Borrowed::new(Self {
+            inner: from_glib_borrow::<_, crate::shared::Shared<_, _>>(ptr).into_inner(),
+            _phantom: std::marker::PhantomData,
+        })
+    }
+}
+#[doc(hidden)]
+impl<'input> FromGlibPtrBorrow<*const ffi::GMatchInfo> for MatchInfo<'input> {
+    #[inline]
+    unsafe fn from_glib_borrow(ptr: *const ffi::GMatchInfo) -> Borrowed<Self> {
+        from_glib_borrow::<_, Self>(ptr as *mut ffi::GMatchInfo)
+    }
+}
+#[doc(hidden)]
+impl<'input> FromGlibContainerAsVec<*mut ffi::GMatchInfo, *mut *mut ffi::GMatchInfo>
+    for MatchInfo<'input>
+{
+    unsafe fn from_glib_none_num_as_vec(ptr: *mut *mut ffi::GMatchInfo, num: usize) -> Vec<Self> {
+        if num == 0 || ptr.is_null() {
+            return Vec::new();
+        }
+        let mut res = Vec::<Self>::with_capacity(num);
+        let res_ptr = res.as_mut_ptr();
+        for i in 0..num {
+            ::std::ptr::write(res_ptr.add(i), from_glib_none(std::ptr::read(ptr.add(i))));
+        }
+        res.set_len(num);
+        res
+    }
+    unsafe fn from_glib_container_num_as_vec(
+        ptr: *mut *mut ffi::GMatchInfo,
+        num: usize,
+    ) -> Vec<Self> {
+        let res = FromGlibContainerAsVec::from_glib_none_num_as_vec(ptr, num);
+        crate::ffi::g_free(ptr as *mut _);
+        res
+    }
+    unsafe fn from_glib_full_num_as_vec(ptr: *mut *mut ffi::GMatchInfo, num: usize) -> Vec<Self> {
+        if num == 0 || ptr.is_null() {
+            crate::ffi::g_free(ptr as *mut _);
+            return Vec::new();
+        }
+        let mut res = Vec::with_capacity(num);
+        let res_ptr = res.as_mut_ptr();
+        ::std::ptr::copy_nonoverlapping(ptr as *mut Self, res_ptr, num);
+        res.set_len(num);
+        crate::ffi::g_free(ptr as *mut _);
+        res
+    }
+}
+#[doc(hidden)]
+impl<'input> FromGlibPtrArrayContainerAsVec<*mut ffi::GMatchInfo, *mut *mut ffi::GMatchInfo>
```

Comment:
I would remove all the array FFI conversion traits for now until someone actually needs them

Reply:
Okay, I removed the `ToGlibContainerFromSlice`/`FromGlibContainerAsVec`/`FromGlibPtrArrayContainerAsVec` impls.